### PR TITLE
ModuleLoader: minor clean-up to remove dead code. NFC

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: lib/Frontend/ModuleInterfaceLoader.cpp [hunk 1]
```diff
@@ -736,13 +737,7 @@ class ModuleInterfaceLoaderImpl {
     // Should we attempt to load a swiftmodule adjacent to the swiftinterface?
     bool shouldLoadAdjacentModule = !ctx.IgnoreAdjacentModules;
 
-    // Don't use the adjacent swiftmodule for frameworks from the public
```
Comment (tshortli) [inline lib/Frontend/ModuleInterfaceLoader.cpp:739]: I personally think this comment is still useful.

File: lib/Frontend/ModuleInterfaceLoader.cpp [hunk 2]
```diff
@@ -736,13 +737,7 @@ class ModuleInterfaceLoaderImpl {
...
-    // Frameworks folder of the SDK.
-    SmallString<128> publicFrameworksPath;
-    llvm::sys::path::append(publicFrameworksPath,
-                            ctx.SearchPathOpts.getSDKPath(),
-                            "System", "Library", "Frameworks");
-    if (isInSystemFrameworks(modulePath)) {
+    if (isInSystemFrameworks(modulePath, true)) {
```
Comment (tshortli) [inline lib/Frontend/ModuleInterfaceLoader.cpp:740]:
```suggestion
    if (isInSystemFrameworks(modulePath, /*publicFramework=*/true)) {
```
Alternatively, you could use an enum to make this value clearer at the call site.

File: lib/Frontend/ModuleInterfaceLoader.cpp (hunk 1 above)
Comment (nkcsgexi) [reply lib/Frontend/ModuleInterfaceLoader.cpp:739]: Added back!

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by tshortli — APPROVED** [review]

**Comment by nkcsgexi** [comment]
@swift-ci Please clean test Linux platform

**Comment by nkcsgexi** [comment]
@swift-ci Please test Windows platform

**Pull request merged**